src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -449,24 +435,24 @@ moduleToJs (Module _ coms mn _ imps exps reExps foreigns decls) foreignInclude =
       done'' <- go remain done'
       js <- binderToJs argVar done'' binder
       return (AST.VariableIntroduction Nothing argVar (Just $ accessorString (mkString $ identToJs field) $ AST.Var Nothing varName) : js)
-  binderToJs' _ _ ConstructorBinder{} =
+  binderToJs _ _ ConstructorBinder{} =
     internalError "binderToJs: Invalid ConstructorBinder in binderToJs"
-  binderToJs' varName done (NamedBinder _ ident binder) = do
+  binderToJs varName done (NamedBinder _ ident binder) = do
     js <- binderToJs varName done binder
     return (AST.VariableIntroduction Nothing (identToJs ident) (Just (AST.Var Nothing varName)) : js)
 
-  literalToBinderJS :: Text -> [AST] -> Literal (Binder Ann) -> m [AST]
-  literalToBinderJS varName done (NumericLiteral num) =
-    return [AST.IfElse Nothing (AST.Binary Nothing AST.EqualTo (AST.Var Nothing varName) (AST.NumericLiteral Nothing num)) (AST.Block Nothing done) Nothing]
-  literalToBinderJS varName done (CharLiteral c) =
-    return [AST.IfElse Nothing (AST.Binary Nothing AST.EqualTo (AST.Var Nothing varName) (AST.StringLiteral Nothing (fromString [c]))) (AST.Block Nothing done) Nothing]
-  literalToBinderJS varName done (StringLiteral str) =
-    return [AST.IfElse Nothing (AST.Binary Nothing AST.EqualTo (AST.Var Nothing varName) (AST.StringLiteral Nothing str)) (AST.Block Nothing done) Nothing]
-  literalToBinderJS varName done (BooleanLiteral True) =
+  literalToBinderJS :: SourceSpan -> Text -> [AST] -> Literal (Binder Ann) -> m [AST]
+  literalToBinderJS ss varName done (NumericLiteral num) =
+    return [AST.IfElse (Just ss) (AST.Binary Nothing AST.EqualTo (AST.Var Nothing varName) (AST.NumericLiteral Nothing num)) (AST.Block Nothing done) Nothing]
+  literalToBinderJS ss varName done (CharLiteral c) =
+    return [AST.IfElse (Just ss) (AST.Binary Nothing AST.EqualTo (AST.Var Nothing varName) (AST.StringLiteral Nothing (fromString [c]))) (AST.Block Nothing done) Nothing]
+  literalToBinderJS ss varName done (StringLiteral str) =
+    return [AST.IfElse (Just ss) (AST.Binary Nothing AST.EqualTo (AST.Var Nothing varName) (AST.StringLiteral Nothing str)) (AST.Block Nothing done) Nothing]
+  literalToBinderJS _ varName done (BooleanLiteral True) =
     return [AST.IfElse Nothing (AST.Var Nothing varName) (AST.Block Nothing done) Nothing]
-  literalToBinderJS varName done (BooleanLiteral False) =
-    return [AST.IfElse Nothing (AST.Unary Nothing AST.Not (AST.Var Nothing varName)) (AST.Block Nothing done) Nothing]
-  literalToBinderJS varName done (ObjectLiteral bs) = go done bs
+  literalToBinderJS ss varName done (BooleanLiteral False) =
+    return [AST.IfElse (Just ss) (AST.Unary Nothing AST.Not (AST.Var Nothing varName)) (AST.Block Nothing done) Nothing]
+  literalToBinderJS _ varName done (ObjectLiteral bs) = go done bs
```

Comment:
Looks like this is what causes the records binders in `case _ of` syntax to no longer have a source span.

Reply:
Oh, that's not what causes that.